### ISSUE TITLE
Crashes after media scanning: IllegalArgumentException:

### ISSUE THREAD

**Issue by fungs** [OxygenCobalt/Auxio#641]
### Describe the Bug/Crash

Start app, see bug report.

### Describe the intended behavior

Not crash.

### What android version do you use?

Android 10

### What device model do you use?

Gigaset GS290 eOS

### Bug report

Crashes at Startup after having completely scanned the library.

java.lang.IllegalArgumentException: Comparison method violates its general contract!
	at java.util.TimSort.mergeLo(TimSort.java:777)
	at java.util.TimSort.mergeAt(TimSort.java:514)
	at java.util.TimSort.mergeCollapse(TimSort.java:439)
	at java.util.TimSort.sort(TimSort.java:245)
	at java.util.Arrays.sort(Arrays.java:1492)
	at java.util.ArrayList.sort(ArrayList.java:1470)
	at java.util.Collections.sort(Collections.java:206)
	at kotlin.collections.CollectionsKt__MutableCollectionsJVMKt.sortWith(Unknown Source:7)
	at org.oxycblt.auxio.list.sort.Sort.songs(SourceFile:21)
	at org.oxycblt.auxio.home.HomeViewModel.onMusicChanges(SourceFile:30)
	at org.oxycblt.auxio.music.MusicRepositoryImpl.dispatchLibraryChange(Unknown Source:27)
	at org.oxycblt.auxio.music.MusicRepositoryImpl$indexImpl$5.invokeSuspend(SourceFile:16)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(Unknown Source:8)
	at kotlinx.coroutines.DispatchedTask.run(Unknown Source:107)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7356)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:940)

### Duplicates

- [X] I have checked the [Troubleshooting](https://github.com/OxygenCobalt/Auxio/wiki/Troubleshooting) page.
- [X] I have checked this issue for duplicates.
- [X] I have checked that this issue occurs on the [lastest version](https://github.com/OxygenCobalt/Auxio/releases).
- [X] I agree to the [Contribution Guidelines](https://github.com/OxygenCobalt/Auxio/blob/dev/.github/CONTRIBUTING.md).

**Comment by OxygenCobalt** [comment]
Have you tried disabling "Intelligent Sorting" in `Settings -> Content`. You should be able to do this before it crashes on a fresh install that's still taking awhile to load music @fungs. This would allow me to confirm what I think it is, that apparently the name comparison code that I do have really does require me to ensure that the comparison and equality results are 100% consistent.

**Comment by fungs** [comment]
@OxygenCobalt, I can confirm that the intelligent sorting (excluding articles etc.) option seems to cause this issue. I disabled it on a fresh install, not running into that issue and enabled it again, which caused the same error.

**Comment by OxygenCobalt** [comment]
Alright, this implies something I feared would happen. The issue is that intelligent name comparison plays a tad loose with the sorting contract. If you have a name like "FooBar" and "Foo Bar", the comparator will consider those equal, but the actual equal function won't. I thought this was okay after reading the comparator contract documentation, but apparently not. Thanks for reporting this @fungs.

**Comment by OxygenCobalt** [comment]
Might have figured out the issue in #810. More or less I was badly abusing Java comparators and violating their invariants with very specific data sequences. Care to try this out @fungs?

[Auxio_Canary.zip](https://github.com/user-attachments/files/16042660/Auxio_Canary.zip)

**Comment by fungs** [comment]
I will give it a try, probably this week!

**Comment by OxygenCobalt** [comment]
Update? @fungs

**Comment by fungs** [comment]
@OxygenCobalt I cleared the cache and data and verified, that the error still exists with the standard version. Then I installed the debug version and got exactly the same error, again. I can reproduce it by clearing cache and data of the app and giving permission to access media at first startup. It scans all media, the loads the media and then shows a message that it could not load the music. I have the option between "try again" and "more". When I select "more", I get the debug log in the first post. When I hit "try again", the music database is empty. When I then restart the app with the empty database, the loading is much faster and the music is there without getting any error.

Could it be that it just has problems parsing a (defective) music file?

I re-did the whole thing many times and also observed that in rare cases I don't get the error at all!
